File: data/src/main/java/com/everis/data/model/FirebaseUserDataMapper.java
package com.everis.data.model;

import com.everis.domain.model.User;
import com.google.firebase.auth.FirebaseUser;

/**
 * Created by everis on 25/04/18.
 */

public class FirebaseUserDataMapper {

    public static User transform(FirebaseUser firebaseUser){
        User user = new User();
        user.setUid(firebaseUser!=null?firebaseUser.getUid():"");
        user.setEmail(firebaseUser!=null?firebaseUser.getEmail():"");
        user.setDisplayName(firebaseUser!=null?firebaseUser.getDisplayName():"");
        return user;
    }

}

File: data/src/main/java/com/everis/data/model/FirebaseUserDataMapper.kt
package com.everis.data.model

import com.everis.domain.model.User
import com.google.firebase.auth.FirebaseUser

object FirebaseUserDataMapper {

    fun transform(firebaseUser: FirebaseUser?): User {
        val user = User()
        user.uid = firebaseUser?.uid ?: ""
        user.email = if (firebaseUser != null) firebaseUser.email else ""
        user.displayName = if (firebaseUser != null) firebaseUser.displayName else ""
        return user
    }

}
